support/src/traits.rs

```diff
@@ -80,3 +85,50 @@ pub trait GetWorstCase {
 pub trait ItemFilter<Item> {
 	fn should_include(&self, credential: &Item) -> bool;
 }
+
+pub trait StorageItemMeter<AccountId, Key> {
+	type Currency: ReservableCurrency<AccountId>;
+
+	fn deposit(
+		key: &Key,
+	) -> Result<Deposit<AccountId, <Self::Currency as Currency<AccountId>>::Balance>, DispatchError>;
+
+	fn deposit_amount(key: &Key) -> <Self::Currency as Currency<AccountId>>::Balance;
+
+	fn store_deposit(
+		key: &Key,
+		deposit: Deposit<AccountId, <Self::Currency as Currency<AccountId>>::Balance>,
+	) -> Result<(), DispatchError>;
+
+	fn change_deposit_owner(key: &Key, new_owner: AccountId) -> Result<(), DispatchError> {
```

Comment:
This does not simply change the deposit owner tho, right? It also potentially updates the deposit amount. I think that should be reflected in the name somehow, as right now it's kinda misleading.

Reply:
the easiest to understand would be that there is one call to update the owner and one to update the amount. I think that's how it should be. If you want both, you batch them together... [da0bdff](https://github.com/KILTprotocol/kilt-node/pull/427/commits/da0bdff5d2d433f2b8e1b849f7a0caebbb9722bb)